pkg/endpoint/bpf.go

```diff
@@ -475,11 +476,18 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir string) error {
 		// Regenerate policy and apply any options resulting in the
 		// policy change.
 		// This also populates e.PolicyMap
-		if _, err = e.regeneratePolicy(owner, nil); err != nil {
+		_, consumersToRm, err := e.regeneratePolicy(owner, nil)
+		if err != nil {
 			e.Mutex.Unlock()
 			return fmt.Errorf("Unable to regenerate policy for '%s': %s",
 				e.PolicyMap.String(), err)
 		}
+		if len(consumersToRm) != 0 {
+			ip4 := e.IPv4.IP()
+			ip6 := e.IPv6.IP()
+			isLocal := e.Opts.IsEnabled(OptionConntrackLocal)
+			go owner.CleanCTEntries(e, isLocal, []net.IP{ip4, ip6}, consumersToRm)
```

Comment:
Why is this done async?

Reply:
If there's an error while cleaning the CT entries, we log that error to the endpoint logger, the logger will need to lock which will cause a deadlock. This might be solved the endpoint logging to be lockless in #1952

Reply:
@tgraf should that be done on all CleanCTEntries?